### PR TITLE
Fix some shape2d get_moment_of_inertia error

### LINKED ISSUE #28842: Mass not used for inertia for CircleShape2D
**Godot version:**
3.1.1

**OS/device including version:**
Ubuntu 18.04

**Issue description:**
The moment of inertia computation doesn't take into account the mass for CircleShape2D.
It seems to be missing in the formula used in CircleShape2DSW::get_moment_of_inertia().
For other Shape2D the moment of inertia grows linearly with the mass.

**Minimal reproduction project:**
[inertia_bug.zip](https://github.com/godotengine/godot/files/3170408/inertia_bug.zip)

### PR DESCRIPTION
**Pull request by Raphael2048**
Fix #28842 and some other shape2d get_moment_of_inertia calculation error
according formula here https://en.wikipedia.org/wiki/List_of_moments_of_inertia

### CODE REVIEW

File: servers/physics_2d/shape_2d_sw.cpp [hunk 1]
```diff
@@ -318,7 +313,9 @@ bool CircleShape2DSW::intersect_segment(const Vector2 &p_begin, const Vector2 &p
 
 real_t CircleShape2DSW::get_moment_of_inertia(real_t p_mass, const Size2 &p_scale) const {
 
-	return (radius * radius) * (p_scale.x * 0.5 + p_scale.y * 0.5);
+	real_t a = radius * p_scale.x;
+	real_t b = radius * p_scale.y;
+	return p_mass * (a * a + b * b) / 4;
```
Comment (aaronfranke) [inline servers/physics_2d/shape_2d_sw.cpp]: Wouldn't `a * a` be the same as `radius * radius * p_scale.x * p_scale.x` then? And `radius * radius` will also happen for `b`. Might be more efficient to calculate that only once with `real_t r2 = radius * radius;`

Comment (raphael10241024) [reply servers/physics_2d/shape_2d_sw.cpp]: Another way is written as `return p_mass * (radius * radius) * p_scale.dot(p_scale) / 4;`. Which needs same calculation times in fact. And I think the origin code looks more easy to understand.

File: servers/physics_2d/shape_2d_sw.cpp [hunk 2]
```diff
@@ -637,7 +634,7 @@ real_t ConvexPolygonShape2DSW::get_moment_of_inertia(real_t p_mass, const Size2
 		aabb.expand_to(points[i].pos * p_scale);
 	}
 
-	return p_mass * aabb.size.dot(aabb.size) / 12.0 + p_mass * (aabb.position + aabb.size * 0.5).length_squared();
```
Comment (akien-mga) [inline servers/physics_2d/shape_2d_sw.cpp]: Are you sure about removing the offset for the center of gravity in this and SegmentShape2D? The formulas on Wikipedia are given for an axis passing through the center of gravity, but maybe that's not what we take as referential in Godot?

Comment (raphael10241024) [reply servers/physics_2d/shape_2d_sw.cpp]: I'm not quite familiar with it. But I think if we need a moment of inertia with an axis not passing center of gravity, we should recalculate it with origin result, there is already it: 				https://github.com/godotengine/godot/blob/252c841d7ff5d770e8d2819e4f7955363410744b/servers/physics_2d/body_2d_sw.cpp#L76
It seems unnecessary to add offset here.

Comment (akien-mga) [reply servers/physics_2d/shape_2d_sw.cpp]: Looks like it indeed.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**